arch/x86/core/intel64/locore.S

```diff
@@ -169,6 +169,30 @@ go64:	movl %cr4, %eax			/* enable PAE and SSE */
 	call z_loapic_enable
 #endif
 
+#ifdef CONFIG_USERSPACE
+	/* Set landing site for system calls made with 'syscall' instruction */
+	movq $z_x86_syscall_entry_stub, %rax
+	movq %rax, %rdx
+	shrq $32, %rdx
+	movl $X86_LSTAR_MSR, %ecx
+	/* LSTAR set to 64-bit address denoted by EDX:EAX */
+	wrmsr
+
+	/* Set segment descriptors in STAR */
+	xorl %eax, %eax			/* Zero low bits, reserved */
+	movl $X86_STAR_UPPER, %edx
+	movl $X86_STAR_MSR, %ecx
+	wrmsr
+
+	/* Set EFLAGS mask applied when making system calls. Currently we
+	 * mask interrupts and clear direction flag.
+	 */
+	movl $0x600, %eax
+	xorl %edx, %edx
+	movl $X86_FMASK_MSR, %ecx
+	wrmsr
```

Comment:
Semi-hard style issue here.  I think this is the wrong place.  go64 is the lowest level bootstrap code required to get us to z_cstart(), these MSRs won't be inspected by the hardware until after the first user thread is spawned much, much later.   Why can't this be done in the C side of initialization with much simpler code, e.g.:

```c
z_x86_msr_write(X86_LSTAR_MSR, z_x86_syscall_entry_stub);
z_x86_msr_write(X86_STAR_MSR, X86_STAR_UPPER << 32);
z_x86_msr_write(X86_FMASK_MSR, 0x600);
```

Reply:
@andyross sure, it's better to do stuff in C if we can. I'll adjust this.